narwhals/_pandas_like/dataframe.py

```diff
@@ -698,21 +698,38 @@ def to_numpy(self, dtype: Any = None, copy: bool | None = None) -> Any:
             # pandas default differs from Polars, but cuDF default is True
             copy = self._implementation is Implementation.CUDF
 
+        to_convert = set()
+        for key, val in self.schema.items():
+            if val == self._dtypes.Datetime and val.time_zone is not None:  # type: ignore[attr-defined]
+                to_convert.add(key)
```

Comment:
I am a big fan of comprehensions, I guess it's just a matter of taste, but if you want:

```suggestion
        to_convert = {
            key for key, val in self.schema.items()
            if val == self._dtypes.Datetime and val.time_zone is not None  # type: ignore[attr-defined]
        } 
```